client/src/components/app.js

```diff
@@ -1,30 +1,42 @@
 import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
+
 import NavBar from './nav-bar';
+import InternalNav from './internal-nav';
 import * as actions from '../data/actions';
+import Footer from './footer';
 
 // TODO: Abstract away app state logic to a Container
 
 const App = (props) => {
-  // Hides main Navbar when in either Whiteboard or Text Editor views
-  const showNavbar = () =>
-    !(props.location.pathname === '/whiteboard' || props.location.pathname === '/texteditor')
-;
+  const notHome = props.location.pathname === '/whiteboard' || props.location.pathname === '/texteditor';
+
+  const showNav = () => (
+    notHome ? <InternalNav location={location} /> : <NavBar />
+  );
+
+  const showFooter = () => (
+    !notHome ? <Footer /> : undefined
+  );
 
   return (
     <div>
-      { showNavbar() && <NavBar props={props} /> }
-      { props.children }
+      <header>
+        { showNav() }
+      </header>
+      <main>
+        { props.children }
+      </main>
+      { showFooter() }
     </div>
   );
 };
 
 App.propTypes = {
-  // showModal: PropTypes.func.isRequired,
+  children: PropTypes.element,
   location: PropTypes.shape({
-    pathname: PropTypes.string.isRequired,
-  }).isRequired,
-  children: PropTypes.node.isRequired,
+    pathname: PropTypes.string,
+  }),
```

Comment:
The previous version of this proptype validation is the preferred method, since `location.pathname` is required.